src/utilities/transition/BaseTransition.js

```diff
@@ -0,0 +1,250 @@
+    _dom.style.mozTransitionDuration = '0';
+    _dom.style.oTransitionDuration = '0';
+    _dom.style.transitionDuration = '0';
+    _dom.removeEventListener( _transitionEndEvent,
+                              _transitionCompleteBinded );
+    _transitionCompleteBinded();
+    _dom.style.webkitTransitionDuration = '';
+    _dom.style.mozTransitionDuration = '';
+    _dom.style.oTransitionDuration = '';
+    _dom.style.transitionDuration = '';
+  }
+
+  /**
+   * Add an event listener to the transition, or call the transition
+   * complete handler immediately if transition not supported.
+   */
+  function _addEventListener() {
+    _isAnimating = true;
+    // If transition is not supported, call handler directly (IE9/OperaMini).
+    if ( _transitionEndEvent ) {
+      _dom.addEventListener( _transitionEndEvent,
+                             _transitionCompleteBinded );
+      this.trigger( BaseTransition.BEGIN_EVENT, { target: this } );
+    } else {
+      this.trigger( BaseTransition.BEGIN_EVENT, { target: this } );
+      _transitionCompleteBinded();
+    }
+  }
+
+  /**
+   * Remove an event listener to the transition.
+   */
+  function _removeEventListener() {
+    _dom.removeEventListener( _transitionEndEvent, _transitionCompleteBinded );
+  }
+
+  /**
+   * Handle the end of a transition.
+   */
+  function _transitionComplete() {
+    _removeEventListener();
+    this.trigger( BaseTransition.END_EVENT, { target: this } );
+    _isAnimating = false;
+  }
+
+  /**
+   * Search for and remove initial BaseTransition classes that have
+   * already been applied to this BaseTransition's target element.
+   */
+  function _flush() {
+    for ( var prop in _classes ) {
+      if ( _classes.hasOwnProperty( prop ) &&
+           _classes[prop] !== _classes.BASE_CLASS &&
+           _dom.classList.contains( _classes[prop] ) ) {
+        _dom.classList.remove( _classes[prop] );
+      }
+    }
+  }
+
+  /**
+   * Remove all transition classes, if transition is initialized.
+   * @returns {boolean}
+   *   True, if the element's CSS classes were touched, false otherwise.
+   */
+  function remove() {
+    if ( _dom ) {
+      halt();
+      _dom.classList.remove( _classes.BASE_CLASS );
+      _flush();
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * @param {string} className - A CSS class.
+   * @returns {boolean} False if the class is already applied
+   *   or the transition is not initialized,
+   *   otherwise true if the class was applied.
+   */
+  function applyClass( className ) {
+    if ( !_dom ) { return false; }
+    if ( !_isFlushed ) {
+      _flush();
+      _isFlushed = true;
+    }
+
+    if ( _dom.classList.contains( className ) ) {
+      return false;
+    }
+
+    _removeEventListener();
+    _dom.classList.remove( _lastClass );
+    _lastClass = className;
+    _addEventListenerBinded();
+    _dom.classList.add( _lastClass );
+
+    return true;
+  }
+
+  // TODO: Update Expandables to use a transition.
```

Comment:
Remove this TODO as it doesn't apply in the context of where this PR is ending up.